pkg/platform/kube/platform.go

```diff
@@ -1768,11 +1768,13 @@ func (p *Platform) enrichFunctionNodeSelector(ctx context.Context, functionConfi
 		"nodeSelector", p.Config.Kube.DefaultFunctionNodeSelector)
 	functionConfig.Spec.NodeSelector = labels.Merge(functionProject.GetConfig().Spec.DefaultFunctionNodeSelector, functionConfig.Spec.NodeSelector)
 
-	p.Logger.DebugWithCtx(ctx,
-		"Enriching function node selector from platform config",
-		"functionName", functionConfig.Meta.Name,
-		"nodeSelector", p.Config.Kube.DefaultFunctionNodeSelector)
-	functionConfig.Spec.NodeSelector = labels.Merge(p.Config.Kube.DefaultFunctionNodeSelector, functionConfig.Spec.NodeSelector)
+	if p.Config.Kube.MergePlatformAndProjectNodeSelectors {
```

Comment:
According to the PR description:
> The default behavior has been modified to consider the platform's NodeSelectors only if the project's NodeSelectors are not set

However, platform's node selector will be ignored if this flag is false, regardless of the project's node selector.
So - what is the real desired behavior?

Reply:
If flag isn't set, we merge platform's and project's selectors (as we do now). If flag is set,  we don't merge them - we check if project node selectors are set and if they are, we merge them with function's ones and ignore platforms. If project's node selectors aren't set, we merge function selector with platform's selectors.